### PR TITLE
app-forensics/aide: verbump to 0.16

### PR DESCRIPTION
**Pull request by Coacher**

### CODE REVIEW

File: app-forensics/aide/aide-0.16.ebuild [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="acl audit curl e2fs mhash postgres prelink selinux static xattr zlib"
+
+COMMON_DEPEND="
+	!mhash? (
+		dev-libs/libgcrypt:0=
+		dev-libs/libgpg-error
+	)
+	mhash? ( app-crypt/mhash )
```
Comment (mgorny) [inline app-forensics/aide/aide-0.16.ebuild]: Sort, plox? I suppose you did that on purpose but having it on top looks weird anyway.

File: app-forensics/aide/aide-0.16.ebuild [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	eautoreconf
+}
+
+src_configure() {
+	local myeconfargs=(
+		--sysconfdir="${EPREFIX}/etc/${PN}"
+		--with-confighmactype="sha512"		# Override default weak MD5 hash.
+		--with-dbhmackey="sha512"			# Override default weak MD5 hash.
+		# Disable broken l10n support: https://sourceforge.net/p/aide/bugs/98/
+		# This doesn't affect anything because there are no localizations yet.
```
Comment (mgorny) [inline app-forensics/aide/aide-0.16.ebuild]: 'yet' sounds funny for software that old.

File: app-forensics/aide/aide-0.16.ebuild [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		--without-locale
+		$(use_enable static)
+		$(use_with zlib)
```
Comment (mgorny) [inline app-forensics/aide/aide-0.16.ebuild]: Do those options have any specific order?

File: app-forensics/aide/files/aide-0.16-respect-configure-options.patch [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+diff --git a/configure.ac b/configure.ac
+index 3598ebe..5aa36cc 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -263,7 +263,7 @@ AC_CHECK_FUNCS(stricmp strnstr strnlen)
+ AC_ARG_WITH([mmap],
+ 	[AC_HELP_STRING([--with-mmap],
+ 		[use mmap @<:@default=check@:>@])],
+-	[],
++	[with_mmap="$withval"],
```
Comment (mgorny) [inline app-forensics/aide/files/aide-0.16-respect-configure-options.patch]: I'm pretty sure this is unnecessary.

https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/External-Software.html

> The option's argument is available to the shell commands action-if-given in the shell variable withval, which is actually just the value of the shell variable with_package, with any - characters changed into ‘_’. You may use that variable instead, if you wish.

**Comment by mgorny** [comment]
Please also cover modernizing the old stable, or at least killing confutils inherit there.

File: app-forensics/aide/aide-0.16.ebuild (hunk 3 above)
Comment (Coacher) [inline app-forensics/aide/aide-0.16.ebuild]: Yes. This is how they are ordered in configure{,.ac}.

File: app-forensics/aide/aide-0.16.ebuild (hunk 2 above)
Comment (Coacher) [inline app-forensics/aide/aide-0.16.ebuild]: It's on their todo.

File: app-forensics/aide/aide-0.16.ebuild (hunk 1 above)
Comment (Coacher) [inline app-forensics/aide/aide-0.16.ebuild]: mhash USE is special so it goes first. The rest is alphabetical.

**Comment by Coacher** [comment]
> Please also cover modernizing the old stable, or at least killing confutils inherit there.

I see no point. It's halfway broken and that codebase is very old. It should just die once 0.16 is stabilized.

File: app-forensics/aide/files/aide-0.16-respect-configure-options.patch (hunk 4 above)
Comment (Coacher) [inline app-forensics/aide/files/aide-0.16-respect-configure-options.patch]: This is needed for posix-acl configure option that maps to acl USE.
Then I went ahead and updated other options with this pattern to match the rest.

Comment (mgorny) [inline app-forensics/aide/files/aide-0.16-respect-configure-options.patch]: `configure` before:

```
# Check whether --with-mmap was given.
if test "${with_mmap+set}" = set; then :
  withval=$with_mmap;
else
  with_mmap=check

fi
```

`configure` after (with your patch):
```
# Check whether --with-mmap was given.
if test "${with_mmap+set}" = set; then :
  withval=$with_mmap; with_mmap="$withval"
else
  with_mmap=check

fi
```

Comment (Coacher) [inline app-forensics/aide/files/aide-0.16-respect-configure-options.patch]: Fixed.

File: app-forensics/aide/files/aide-0.16-fix-acl-configure-option.patch [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+diff --git a/configure.ac b/configure.ac
+index 3598ebe..c45bbee 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -450,7 +450,7 @@ AC_MSG_CHECKING(for posix-acl-support)
+ AC_ARG_WITH([posix-acl],
+ 	[AC_HELP_STRING([--with-posix-acl],
+ 		[use POSIX ACLs (no checking)])],
+-	[],
++	[with_posix_acl_support="$withval"],
```
Comment (mgorny) [inline app-forensics/aide/files/aide-0.16-fix-acl-configure-option.patch]: Alternative suggestion: maybe just update other places in the code to use `$with_posix_acl` variable as defined by autoconf? ;-)

File: app-forensics/aide/metadata.xml [hunk 6]
```diff
@@ -1,12 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
+	<maintainer type="person">
+		<email>itumaykin+gentoo@gmail.com</email>
+		<name>Coacher</name>
+	</maintainer>
 	<maintainer type="project">
-		<email>forensics@gentoo.org</email>
```
Comment (mgorny) [inline app-forensics/aide/metadata.xml]: Oh, please do not remove forensics. Keeping it is safer politically ;-).

File: app-forensics/aide/files/aide-0.16-fix-acl-configure-option.patch (hunk 5 above)
Comment (Coacher) [inline app-forensics/aide/files/aide-0.16-fix-acl-configure-option.patch]: No.

File: app-forensics/aide/metadata.xml (hunk 6 above)
Comment (Coacher) [inline app-forensics/aide/metadata.xml]: Fixed.

Comment (Coacher) [inline app-forensics/aide/metadata.xml]: And yes, I don't want to be a primary maintainer at least for now.

**Comment by Coacher** [comment]
Thank you, @mgorny.

**Pull request closed without merging**